### PR TITLE
NET-5186 Add NET_BIND_SERVICE capability to Consul's restricted securityContext

### PR DESCRIPTION
**Pull request by nathancoleman**
Changes proposed in this PR:
- Add the `NET_BIND_SERVICE` capability for consul-dataplane related to https://github.com/hashicorp/consul-dataplane/pull/238

How I've tested this PR:
- Install consul w/ ingress-gateway serving on a privileged port such as `443`
- Install consul under restricted pod security policy
- Install consul under restricted[-v2] OpenShift security context constraints

How I expect reviewers to test this PR:
- See above

Checklist:
- [ ] Tests added
- [ ] [CHANGELOG entry added](https://github.com/hashicorp/consul-k8s/blob/main/CONTRIBUTING.md#adding-a-changelog-entry)

### CODE REVIEW

**Review by david-yu — APPROVED** [review]
This sounds like the best solution to allow privileged ports for now. Should work with both GKE Autopilot and OpenShift.

File: .changelog/2787.txt [hunk 1]
```diff
@@ -0,0 +1,3 @@
+```release-note:security
+Add NET_BIND_SERVICE capability to restricted security context used for consul-dataplane
```
Comment (nathancoleman) [inline .changelog/2787.txt:2]: @david-yu do you have thoughts on how this appears in the changelog? I was torn between `improvement` and `security` as the not category and wasn't sure if it made sense to include any more details here or a link to the related consul-dataplane PR which requires this change.

Comment (david-yu) [reply .changelog/2787.txt:2]: I believe this is more of an improvement than a security change which is typically reserved for addressing CVEs.

Comment (nathancoleman) [reply .changelog/2787.txt:2]: Updated

**Review by curtbushko — APPROVED** [review]
Thanks Nathan!

**Pull request merged**